client/signup/steps/plans/tabbed-plans.js

```diff
@@ -0,0 +1,1316 @@
+	const bestForStrings = getBestForStrings();
+
+	const [ featureComparison, setFeatureComparison ] = useState(
+		featureComparisonData.Professional
+	);
+	const [ planDetails, setPlanDetails ] = useState();
+	const [ selectedTab, setSelectedTab ] = useState( tabList[ 1 ] );
+	const [ termLength, setTermLength ] = useState( 'annually' );
+	const [ showFeatures, setShowFeatures ] = useState( false );
+	const [ primaryButton, setPrimaryButton ] = useState( 'Premium' );
+	const dispatch = useDispatch();
+
+	const toggleTab = () => {
+		selectedTab === tabList[ 0 ] ? setSelectedTab( tabList[ 1 ] ) : setSelectedTab( tabList[ 0 ] );
+	};
+
+	const toggleTerm = () =>
+		termLength === 'annually' ? setTermLength( 'monthly' ) : setTermLength( 'annually' );
+
+	const handleUpgradeButtonClick = ( productSlug, productId ) => {
+		const args =
+			productSlug === 'free_plan' ? null : { product_slug: productSlug, product_id: productId };
+
+		dispatch(
+			recordTracksEvent( 'calypso_signup_plans_page_clicks', {
+				plan: productSlug,
+			} )
+		);
+
+		onUpgradeClick( args );
+	};
+
+	const tabHandler = ( event ) => {
+		if ( event.code === 'Tab' ) {
+			event.preventDefault();
+			toggleTab();
+		}
+	};
+
+	useEffect( () => {
+		const planFilter =
+			selectedTab === 'Professional'
+				? [ 'Premium', 'Business', 'eCommerce' ]
+				: [ 'Free', 'Personal' ];
+
+		const displayedPlans = planProperties
+			.filter( ( plan ) => planFilter.includes( plan.planName ) )
+			.filter( ( plan ) => plan.termLength === termLength || plan.termLength === null );
+
+		setFeatureComparison( featureComparisonData[ selectedTab ] );
+		setPlanDetails( displayedPlans );
+	}, [ selectedTab, planProperties, termLength ] );
+
+	useEffect( () => {
+		selectedTab === tabList[ 0 ] ? setPrimaryButton( 'Personal' ) : setPrimaryButton( 'Business' );
+		dispatch(
+			recordTracksEvent( 'calypso_signup_plans_page_clicks', {
+				tab: selectedTab,
+			} )
+		);
+		window.addEventListener( 'keydown', tabHandler );
+		return () => {
+			window.removeEventListener( 'keydown', tabHandler );
+		};
+	}, [ selectedTab ] );
+
+	useEffect( () => {
+		dispatch(
+			recordTracksEvent( 'calypso_signup_plans_page_clicks', {
+				term: termLength,
+			} )
+		);
+	}, [ termLength ] );
+
+	return (
+		<>
+			<TabWrapper>
+				<Tabs>
+					{ tabList.map( ( item, index ) =>
+						item === selectedTab ? (
+							<SelectedTab
+								key={ `tab${ index }` }
+								className={ `tabbed-plans__tab-${ index + 1 }` }
+								onClick={ toggleTab }
+							>
+								{ item }
+							</SelectedTab>
+						) : (
+							<Tab
+								key={ `tab${ index }` }
+								className={ `tabbed-plans__tab-${ index + 1 }` }
+								onClick={ toggleTab }
+							>
+								{ item }
+							</Tab>
+						)
+					) }
+				</Tabs>
+			</TabWrapper>
+			<Grid className="tabbed-plans__grid-container">
+				{ selectedTab === 'Professional' && (
+					<TermToggles className="tabbed-plans__term-toggles">
+						<RadioButton>
```

Comment:
I found it quite confusing that `RadioButton` component is actually just a label underneath. I think this nested component structure:
```
<RadioButton>
							<input
								type="radio"
								checked={ termLength === 'monthly' ? 'checked' : '' }
								name="monthly"
								onChange={ toggleTerm }
							/>
							<Checkmark></Checkmark>
							<span>{ 'Pay monthly' }</span>
						</RadioButton>
```
already worths to be extracted as one component. By doing so we can also eliminate code duplication of writing the same nested component structure for twice here.

Reply:
Changed that to RadioButtonLabel to be more clear. I'll add this to the refactor as well. Let me know if you'd rather see everything DRY'd up for this release and I'll knock it out Monday.

Reply:
Let's iterate on it in the follow-up PR then.  :)

Reply:
I ended up refactoring this into a separate component to meet a change in spec (p2 post coming from @victorespigares  soon to describe the changes).